GETTING_STARTED_CLI.markdown

```diff
@@ -0,0 +1,294 @@
+[2021-02-12T00:55:45Z INFO  actix_web::middleware::logger] 127.0.0.1:54768 "POST /SGX/Msg3 HTTP/1.1" 200 25 "-" "-" 0.695685
+[2021-02-12T00:55:45Z INFO  actix_server::builder] Starting 12 workers
+[2021-02-12T00:55:45Z INFO  actix_server::builder] Starting "actix-web-service-127.0.0.1:3017" service on 127.0.0.1:3017
+[2021-02-12T00:55:45Z INFO  veracruz_server] Veracruz Server running on 127.0.0.1:3017
+```
+
+## Running the Veracruz Client 
+
+In practice you may integrate the `veracruz-client` Rust crate directly into
+your application, sending data programatically via the Rust API, but for this
+example we'll use the Veracruz Client CLI interface. This provides the same
+functionality, but in a CLI form.
+
+First lets send over the program to our Veracruz server, this requires an
+identity with the "PiProvider" role:
+
+<!-- TODO do we need - - target sgx? -->
+
+``` bash
+$ vc-client example/example-policy.json \
+    --target sgx \
+    --key example/example-program-key.pem \
+    --identity example/example-program-cert.pem \
+    --program example-binary.wasm=example/example-binary.wasm
+Loaded policy example/example-policy.json 645ae94ea86eaf15cfc04c07a17bd9b6a3b3b6c3558fae6fb93d8ee4c3e71241
+Connecting to 127.0.0.1:3017
+Submitting <enclave>/example-binary.wasm from example/example-binary.wasm
+```
+
+Then lets send over our datas with identities with the "DataProvider" role.
+These are shares I've created that the example will use to reconstruct a
+secret message. Keep in mind most likely these datas will be coming from
+different devices:
+
+``` bash
+$ vc-client example/example-policy.json \
+    --target sgx \
+    --key example/example-data0-key.pem \
+    --identity example/example-data0-cert.pem \
+    --data input-0=<(echo "01dc061a7bdaf77616dd5915f3b4" | xxd -r -p)
+Loaded policy example/example-policy.json 645ae94ea86eaf15cfc04c07a17bd9b6a3b3b6c3558fae6fb93d8ee4c3e71241
+Connecting to 127.0.0.1:3017
+Submitting <enclave>/input-0 from /dev/fd/63
+
+$ vc-client example/example-policy.json \
+    --target sgx \
+    --key example/example-data1-key.pem \
+    --identity example/example-data1-cert.pem \
+    --data input-1=<(echo "027f38e27b5a02a288d064965364" | xxd -r -p)
+Loaded policy example/example-policy.json 645ae94ea86eaf15cfc04c07a17bd9b6a3b3b6c3558fae6fb93d8ee4c3e71241
+Connecting to 127.0.0.1:3017
+Submitting <enclave>/input-1 from /dev/fd/63
+
+$ vc-client example/example-policy.json \
+    --target sgx \
+    --key example/example-data2-key.pem \
+    --identity example/example-data2-cert.pem \
+    --data input-2=<(echo "03eb5b946cefd583f17f51e781da" | xxd -r -p)
+Loaded policy example/example-policy.json 645ae94ea86eaf15cfc04c07a17bd9b6a3b3b6c3558fae6fb93d8ee4c3e71241
+Connecting to 127.0.0.1:3017
+Submitting <enclave>/input-2 from /dev/fd/63
+```
+
+And finally, we can request the result using an identity with the
+"RequestResult" role:
+
+``` bash
+$ vc-client example/example-policy.json \
```

Comment:
how would `vc-client` handle a case where the identity had more than one role?

Reply:
I don't have this in the example, but vc-client handles this. You just need to provide multiple flags to vc-client. The operations will always be in program->data->result->shutdown order

Reply:
I would prefer not to have ordering determine the operation type implicitly. Better to have an OPCODE flag (IMHO).

Reply:
Sorry, I wasn't clear

``` bash
$ vc-client policy.json \
    --identity me-cert.pem \
    --key me-key.pem \
    --program program.wasm \
    --data data1.txt \
    --data data2.txt \
    --result result.out
```

Would result in uploading program.wasm, data1.txt, data2.txt, running the calculation, and reading the result into result.out IFF me-cert.pem has the permissions in the policy to do so.

(Actually right now I think it tries these in order and fails part of the way through, I can add a check so vc-client bails if the request does not match the policy, that would help catch user mistakes)

The operations always occur in program -> data -> result -> shutdown order, because that's sort of the only order that makes sense (reordering data/program doesn't have an effect), and I don't think the order of command line _flags_ are usually expected to have an effect on the operation.